RecoLocalMuon/GEMSegment/plugins/ME0SegmentAlgorithm.cc

```diff
@@ -44,21 +44,21 @@ ME0SegAlgoMM::ME0SegAlgoMM(const edm::ParameterSet& ps) : ME0SegmentAlgorithm(ps
 /* Destructor
  *
  */
-ME0SegAlgoMM::~ME0SegAlgoMM() {
+ME0SegmentAlgorithm::~ME0SegmentAlgorithm() {
 }
 
 
-std::vector<ME0Segment> ME0SegAlgoMM::run(const ME0Ensemble& ensemble, const EnsembleHitContainer& rechits) {
+std::vector<ME0Segment> ME0SegmentAlgorithm::run(const ME0Ensemble& ensemble, const EnsembleHitContainer& rechits) {
 
   theEnsemble = ensemble;
```

Comment:
this is a very heavy weight object to copy and then not change (as far as I see).Could you instead pass it down to the routine that needs it as a const reference?